content/blog/peace-of-mind-with-cloud-secret-providers/index.md

```diff
@@ -0,0 +1,668 @@
+    Statement: [
+        // This statement allows all users to view the key in the console
+        {
+            Sid: "AllowGetKeys",
+            Effect: "Allow",
+            Action: ["kms:Describe*", "kms:Get*", "kms:List*"],
+            Principal: {
+                "AWS": [`arn:aws:iam::${accountID}:root`]
+            },
+            Resource: "*",
+        },
+        // This is a configurable statement that we can use to allow access to an IAM arn
+        {
+            Sid: "AllowIAMUserAccessKeys",
+            Effect: "Allow",
+            Action: ["kms:*"],
+            Principal: {
+                "AWS": iamRole,
+            },
+            Resource: "*",
+        }
+    ]
+}
+
+// Create a nw KMS Key
+const key = new aws.kms.Key("stack-encryption-key", {
+        deletionWindowInDays: 10,
+        description: "KMS key for encrypting Pulumi secret values",
+        policy: JSON.stringify(keyPolicy),
+});
+
+
+// Create an alias to the key
+const alias = new aws.kms.Alias("alias/stack-encryption-key", {
+    targetKeyId: key.keyId
+});
+
+
+// Export the arns
+exports.keyArn = key.arn;
+exports.aliasArn = alias.arn;
+
+});
+
+
+// Export the arns
+exports.keyArn = key.arn;
+exports.aliasArn = alias.arn;
+```
+{{% /choosable %}}
+
+{{% choosable language python %}}
+```python
+import pulumi
+import json
+from pulumi_aws import kms
+
+config = pulumi.Config()
+iam_role = config.require("iamRole")
+account_id = config.require("accountID")
+
+
+key_policy = {
+    "Version": "2012-10-17",
+    "Id": "policy",
+    "Statement": [
+        # This statement allows all users to view the key in the console
+        {
+            "Sid": "AllowGetKeys",
+            "Effect": "Allow",
+            "Action": ["kms:Describe*", "kms:Get*", "kms:List*"],
+            "Principal": {
+                "AWS": ["arn:aws:iam::{}:root".format(account_id)]
+            },
+            "Resource": "*",
+        },
+        # This is a configurable statement that we can use to allow access to an IAM arn
+        {
+            "Sid": "AllowIAMUserAccessKeys",
+            "Effect": "Allow",
+            "Action": ["kms:*"],
+            "Principal": {
+                "AWS": iam_role,
+            },
+            "Resource": "*",
+        }
+    ]
+}
+# Create an AWS KMS key
+key = kms.Key("stack-encryption-key",
+    deletion_window_in_days=10,
+    description="KMS key for encrypting Pulumi secret values",
+    policy=json.dumps(key_policy)
+)
+
+alias = kms.Alias("alias/stack-encryption-key",
+    target_key_id=key.key_id
+)
+
+# Export the name of the bucket
+pulumi.export('key_arn',  key.arn)
+pulumi.export('alias_arn', alias.arn)
+```
+{{% /choosable %}}
+
+{{% choosable language go %}}
+
+```go
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/pulumi/pulumi-aws/sdk/v2/go/aws/kms"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi"
+	"github.com/pulumi/pulumi/sdk/v2/go/pulumi/config"
+)
+
+func main() {
+
+	type Principal struct {
```

Comment:
So I *LOVE* these - but this shows a limitation in our core support - we should go with the standard JSON style building here IMO and we can open an issue around adding this overlay

$0.02

Reply:
I'm in two minds about this. I actually think the `json` string inside a `fmt` call looks horrible. I initially wrote it this way and got sick of debugging the errors/trying to get it right but I think it might be personal preference. Anyone else fancy chiming in? I'm not opposed to changing it...

Reply:
Yeah - though very verbose - I actually think the style as written feels more Go-native than doing something with raw strings.  Can't wait to provide all the types out of the box!  (hopefully soon?)

Reply:
Will leave this as it is!